src/lib.rs

```diff
@@ -133,183 +114,114 @@ impl File {
     }
 
     /// Gets the wrapped file.
-    pub fn file(&self) -> &std::fs::File {
+    pub fn file(&self) -> &fs::File {
         &self.file
     }
 
     /// Gets the path of the wrapped file.
     pub fn path(&self) -> &Path {
         &self.path
     }
-
-    fn try_exec<R>(
-        &self,
-        op: impl FnOnce(&std::fs::File) -> std::io::Result<R>,
-        context: impl FnOnce(&path::Display) -> String,
-    ) -> Result<R, Error> {
-        match op(&self.file) {
-            Ok(result) => Ok(result),
-            Err(error) => Err(Error::new(error, context(&self.path.display()))),
-        }
-    }
 }
 
 impl Read for File {
     fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
-        (&mut &*self).read(buf)
+        self.file
+            .read(buf)
+            .map_err(|source| Error::new(source, &self.path))
     }
 
     fn read_vectored(&mut self, bufs: &mut [std::io::IoSliceMut<'_>]) -> std::io::Result<usize> {
-        (&mut &*self).read_vectored(bufs)
+        self.file
+            .read_vectored(bufs)
+            .map_err(|source| Error::new(source, &self.path))
     }
 }
 
 impl<'a> Read for &'a File {
     fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
-        self.try_exec(
-            |mut file| file.read(buf),
-            |path| format!("failed to read from file `{}`", path),
-        )
-        .map_err(Error::into_io_error)
+        (&(**self).file)
```

Comment:
I only learned that there are trait implementations for `&File` today and I struggled to get this re-borrow to work. Is there a cleaner way to do this that you know?